### PR TITLE
Brick

### PR DESCRIPTION
**Pull request by pauperboards**
## Description

This change is to revert ownership and maintenance of the Brick keyboard to me, its designer and developer.  It is a breaking change that moves the firmware directory under keyboards/pauperboards, but also corrects some non-code information in the various files therein. It also tries to move to a more modern implementation, using json.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pauperboards/brick/keymaps/default/keymap.c [hunk 1]
```diff
@@ -1,19 +1,3 @@
...
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
 #include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/pauperboards/brick/keymaps/default/keymap.c:1]: Must have GPL compliant headers (which you have removed?)

File: users/pauper/config.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/*
```
Comment (zvecr) [inline users/pauper/config.h:1]: Userspace additions should be submitted separately

**Review by zvecr — CHANGES_REQUESTED** [review]
>  It also tries to move to a more modern implementation, using json.

Nothing extra has been migrated to json?

File: keyboards/pauperboards/brick/readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    * Brick v0.2 - same layout and components as v0.1, but with the bodge fixed. This version was run as part of Cervello group buy by [P3DStore](http://p3dstore.com).
+* Hardware Availability: Some remaining pcbs might be available for purchase from [P3D](http://p3dstore.com/collections/pcbs/products/cervello-pcb-only-group-buy).
+
+Make example for this keyboard (after setting up your build environment):
+
+    make pauper/brick:default
+
+Flashing example for this keyboard:
+    
+    make pauper/brick:default:flash
```
[the remark this answers is not in the document]
Comment (pauperboards) [reply keyboards/pauperboards/brick/readme.md:20]: Fixing in next push.

File: keyboards/pauperboards/brick/readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Brick
+
+![Brick, top view](https://i.imgur.com/BLJ6Tum.jpg)
+![Brick, layout options](https://i.imgur.com/JhsWzY7.png)
```
[the remark this answers is not in the document]
Comment (pauperboards) [reply keyboards/pauperboards/brick/readme.md:4]: Fixed in next push.

File: keyboards/pauperboards/brick/keymaps/default/keymap.c [hunk 5]
```diff
@@ -1,19 +1,3 @@
...
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
```
[the remark this answers is not in the document]
Comment (pauperboards) [reply keyboards/pauperboards/brick/keymaps/default/keymap.c:16]: Inadvertent. Will re-add.

File: keyboards/pauperboards/brick/keymaps/default/keymap.c (hunk 1 above)
Comment (pauperboards) [reply keyboards/pauperboards/brick/keymaps/default/keymap.c:1]: This was accidental, and will be fixed in the next push.

File: users/pauper/config.h (hunk 2 above)
Comment (pauperboards) [reply users/pauper/config.h:1]: Accident. Thanks for the heads-up. I'll pull it out.

**Comment by pauperboards** [comment]
> Please also update `data/mappings/keyboard_aliases.hjson` per the PR checklist.

Missed that. Added.

**Comment by pauperboards** [comment]
> > It also tries to move to a more modern implementation, using json.
> 
> Nothing extra has been migrated to json?

My changes got lost when I was migrating. I'll fix and push the new version.

File: keyboards/pauperboards/brick/keymaps/default/keymap.c [hunk 6]
```diff
@@ -62,13 +46,58 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+
+    }
+
+    if (IS_LED_ON(usb_led, USB_LED_KANA)) {
+
+    } else {
+
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (pauperboards) [reply keyboards/pauperboards/brick/keymaps/default/keymap.c]: Removed suggested and updated the required json.

File: keyboards/pauperboards/brick/keymaps/default/keymap.c (hunk 1 above)
Comment (pauperboards) [reply keyboards/pauperboards/brick/keymaps/default/keymap.c]: I restored my json implementation.

File: users/pauper/config.h (hunk 2 above)
Comment (pauperboards) [reply users/pauper/config.h]: Made all requested changes.

**Comment by keyboard-magpie** [comment]
I appreciate you may have designed and developed this keyboard but this probably needs sign-off from @JunoNgx to go ahead.

**Comment by pauperboards** [comment]
> I appreciate you may have designed and developed this keyboard but this probably needs sign-off from @JunoNgx to go ahead.

I also wrote the firmware. My copyright was stripped off and replaced.

File: keyboards/pauperboards/brick/config.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
+/* Copyright 2023  Jason Chestnut <pauperboards@gmail.com> @pauperboards
```
Comment (zvecr) [inline keyboards/pauperboards/brick/config.h:1]: This file should be removed

File: keyboards/pauperboards/brick/info.json [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+        "command": false,
+        "console": false,
+        "encoder": true,
+        "extrakey": true,
+        "midi": false,
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true,
+        "unicode": false,
+        "lto": true
```
Comment (zvecr) [inline keyboards/pauperboards/brick/info.json:32]:
```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "encoder": true,
        "rgblight": true,
        "lto": true
```

File: keyboards/pauperboards/brick/info.json [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+    },
+    "encoder": {
+        "rotary": [
+            { "pin_a": "F4", "pin_b": "F5", "resolution": 1 }
```
Comment (zvecr) [inline keyboards/pauperboards/brick/info.json:36]: Some doubt here about a default resolution value of 1.

**Review by zvecr — CHANGES_REQUESTED** [review]
* via keymap still broken
* `keyboards/pauperboards/brick/rules.mk` needs to match template <https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/rules.mk>

**Comment by keyboard-magpie** [comment]
Have been discussing this amongst the collaborators. Can see your personal gmail on one of the first commits for Brick. Can I ask that you email me from your personal gmail and your pauperboards gmail and I think we can get this sorted. 

@vanillakeyboards this is being dealt with here, for reference.

**Comment by keyboard-magpie** [comment]
Right. Have now verified both pauper's email addresses so once requested changes are met I'd be happy to merge this.

File: keyboards/pauperboards/brick/info.json (hunk 9 above)
Comment (pauperboards) [reply keyboards/pauperboards/brick/info.json:36]: Good catch. I fixed this.

File: keyboards/pauperboards/brick/config.h (hunk 7 above)
Comment (pauperboards) [reply keyboards/pauperboards/brick/config.h:1]: Done.

**Comment by pauperboards** [comment]
> 

I completely missed the broken via keymap. Fixed.

**Review by keyboard-magpie — APPROVED** [review]

**Comment by pauperboards** [comment]
@zvecr Can you re-review and approve? I think I addressed all your comments.

**Review by drashna — APPROVED** [review]

**Comment by JunoNgx** [comment]
Late input from me, but sorry for having mishandled the copyright notice and glad to hand over this to the original creator @pauperboards 🙏

**Comment by pauperboards** [comment]
Thank you for that. I should apologize to you and anyone who was frustrated
by the lack of qmk support. I promise to stay on top of it.

**Pull request merged**